platform/src/ActivityManager.js

```diff
@@ -346,15 +346,35 @@ class ActivityManager {
         return null;
     }
 
+    /**
+     * Interpolate someString using information from session storage.
+     * 
+     * @param {*} someString the string to be changed
+     */
+    interpolate(someString) {
+        let result = someString;
+
+        for (let i = 0; i < sessionStorage.length; i++) {
+            let currentKey = sessionStorage.key(i);
 
+            if (currentKey !== "isAuthenticated") {
+                // TODO: This *assumes* this can only be a panel ID, but that may change over time,
+                // so this code may need to be improved to only allow access to panel IDs explicitly
+                result = result.replace("${ID-" + currentKey + "}", sessionStorage.getItem(currentKey));
```

Comment:
Using `${...}` sytanx overlaps with bash and would prevent envsubst being used on activity files consider using an alternative e.g. `#{...}`. We don't currently use envsubst in this way.

Reply:
Good point. I've changed this to `{{ ... }}` in line with standard templating engines

Reply:
LGTM